addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/ws/IhcClient.java

```diff
@@ -536,39 +536,38 @@ private void mysleep(long milli) {
     }
 
     private void sendErrorEvent(IhcExecption err) {
-        try {
-            Iterator<IhcEventListener> iterator = eventListeners.iterator();
+        Iterator<IhcEventListener> iterator = eventListeners.iterator();
 
-            while (iterator.hasNext()) {
+        while (iterator.hasNext()) {
+            try {
                 iterator.next().errorOccured(err);
+            } catch (RuntimeException e) {
+                logger.debug("Event listener invoking error. ", e);
             }
-        } catch (Exception e) {
-            logger.debug("Event listener invoking error", e);
         }
     }
 
     private void sendControllerStateUpdateEvent(WSControllerState state) {
-        try {
-            Iterator<IhcEventListener> iterator = eventListeners.iterator();
+        Iterator<IhcEventListener> iterator = eventListeners.iterator();
```

Comment:
Iterators are ancient technology (Java 1.4) that I would personally only use nowadays in rare cases where items are removed from a collection while iterating over these same items.

Most of the iterator usages can be easily replaced with the so called [enhanced for loop](https://blogs.oracle.com/corejavatechtips/using-enhanced-for-loops-with-your-classes) which already got introduced way back in Java 1.6.

```java
        for (IhcEventListener listener : eventListeners) {
            try {
                listener.statusUpdateReceived(state);
            } catch (Exception e) {
                logger.debug("Event listener invoking error", e);
            }
        }
```

These result in less code and are easier to read/maintain.

Nowadays most of these enhanced for loops can also be replaced with stream operations resulting in even less code.

Reply:
That's all true, but is not part of this feature PR. Do you want to have a look @paulianttila or should we merge?

Reply:
I will do something about that later on, same time than scheduler changes proposed by @Hilbrand.

Reply:
Very good! It doesn't have to be part of this PR. That's why I made a simple comment and not requested changes for this. :wink: